src/db/tables/subscribers.js

```diff
@@ -271,6 +272,59 @@ async function deleteResolutionsWithEmail (id, email) {
 }
 /* c8 ignore stop */
 
+/**
+ * @param {Partial<{ plusOnly: boolean; limit: number; }>} options
+ * @returns {Promise<import("knex/types/tables").SubscriberRow[]>}
+ */
+// Not covered by tests; mostly side-effects. See test-coverage.md#mock-heavy
+/* c8 ignore start */
+async function getSubscribersWaitingForMonthlyEmail (options = {}) {
+  // I'm explicitly referencing the type here, so that these lines of code will
+  // show up as errors when we remove it from the flag list:
+  /** @type {import("./featureFlags.js").FeatureFlagName} */
+  const featureFlagName = "MonthlyActivityEmail";
+  const flag = await knex("feature_flags")
+    .first()
+    .where("name", featureFlagName);
+
+  if (!flag?.is_enabled) {
+    return [];
+  }
+
+  let query = knex('subscribers')
+    .select()
+    .where((builder) => builder.whereNull("monthly_email_optout").orWhere("monthly_email_optout", false))
+    .andWhere(builder => builder.whereNull("monthly_email_at").orWhereRaw('"monthly_email_at" < NOW() - INTERVAL \'30 days\''));
+
+  if (Array.isArray(flag.allow_list) && flag.allow_list.length > 0) {
+    // The `.andWhereIn` alias doesn't exist:
+    // https://github.com/knex/knex/issues/1881#issuecomment-275433906
+    query = query.whereIn("primary_email", flag.allow_list)
+  }
+
+  if (options.plusOnly) {
+    // Note: This will only match people who had a Plus subscription the last
+    //       time they signed in. We'd have to check with SubPlat directly to
+    //       find out if they have a subscription right now.
```

Comment:
The Monitor database is [updated by webhook](https://github.com/mozilla/blurts-server/blob/288ce9797d87d19c3bf7f1a48ca2de9adb246e60/src/app/api/v1/fxa-rp-events/route.ts) as well, so we don't necessarily need to wait until the user logs in again.

This comment *is* relevant for dev environments that don't have webhook handling set up e.g. `localhost`.

Reply:
Oh right, good one - clarified the comment in 9c81d4737.